### PR TITLE
Adds metrics to track awesomebar performance

### PR DESCRIPTION
**Pull request by tarikeshaq**
Adds performance metrics to track awesomebar performance.

This is ahead of the work to move the history implementation to application services, to make sure we can track any performance implications.

~~## Keeping as a draft as a I open the data review~~

### CODE REVIEW

**Comment by tarikeshaq** [comment]
https://github.com/mozilla-mobile/firefox-ios/issues/11602 is the data review, I'll hold off until that's approved then update this PR to link to it in the metrics.yaml

**Comment by tarikeshaq** [comment]
data review was done in https://github.com/mozilla-mobile/firefox-ios/issues/11602!

**Comment by tarikeshaq** [comment]
Adding more context to what we are measuring here - The metrics will track a timing distribution of the time it takes to query the bookmarks and history local stores for the awesome bar, specifically the timer starts at the beginning of the query and stops whenever the query is done for **both** stores. 

This is particularly relevant now because we are replacing the history storage and syncing with the application services implementation. We want to make sure we have a way to track down the performance of the queries

**Review by OrlaM — APPROVED** [review]
LGTM

**Comment by tarikeshaq** [comment]
Thanks @OrlaM! I don't have permission but this should be ready to land - given it's only metrics it should be safe to land in 105 - that said, I wouldn't mind having it land later

**Pull request merged**
